src/videojs-contrib-hls.js

```diff
@@ -247,8 +247,25 @@ class HlsHandler extends Component {
       this.options_.blacklistDuration = 5 * 60;
     }
 
-    // start playlist selection at a reasonable bandwidth for
-    // broadband internet (0.5 MB/s) or mobile (0.0625 MB/s)
+    if (typeof this.options_.bandwidth !== 'number') {
+      if (this.options_.useBandwidthFromLocalStorage && window.localStorage) {
+        const storedBandwidth =
+          window.localStorage.getItem('videojs-contrib-hls-bandwidth');
+        const storedThroughput =
+          window.localStorage.getItem('videojs-contrib-hls-throughput');
+
+        if (storedBandwidth) {
+          this.options_.bandwidth = parseInt(storedBandwidth, 10);
+        }
+
+        if (storedThroughput) {
+          this.options_.throughput = parseInt(storedThroughput, 10);
+        }
+      }
+    }
+
+    // if bandwidth was not set by options or pulled from local storage, start playlist
+    // selection at a reasonable bandwidth
     if (typeof this.options_.bandwidth !== 'number') {
```

Comment:
It's semantically equivalent, but I'm wondering why you didn't include this as an `else` in the clause above. i.e. if use local storage, grab it, otherwise set the bandwidth from a constant.

Reply:
It's possible that the user did not specify a bandwidth value and set `useBandwidthFromLocalStorage` to true, but there was nothing in local storage, leaving bandwidth undefined.

Reply:
ah good call